### DISCUSSION TITLE
add option to skip tls certificate verification

### DISCUSSION THREAD

**Discussion by nortalrwolf** [getseabird/seabird#48] [Ideas]
Just like the title says, please add the option to skip tls verification.
- cluster:
    insecure-skip-tls-verify: true

**Comment by jgillich** [comment]
Would you mind explaining why you need this? IMO this is potentially harmful, usually clusters will either have a valid cert or include their self-signed CA in the kubeconfig.

**Reply by nortalrwolf** [reply]
I would also agree on that this is rather unusual, but it may happen, that with certain proxy setups for the api server or when it would only be exposed to internal networks, that this layer of security might not be needed/wanted.

**Comment by rasschaert** [comment]
I want to chime in here. I'm using a ZTA tunneling solution to securely connect to my cluster. 
It's conceptually similar to using an SSH tunnel, but without keys or passwords.

As a result my kubeconfig is points to localhost.

```
clusters:
- cluster:
    insecure-skip-tls-verify: true
    server: https://127.0.0.1:4488
```

Including the certificate in the kubeconfig doesn't work, since it doesn't match the hostname, resulting in this error with just about every tool, including kubectl:

> Unable to connect to the server: tls: failed to verify certificate: x509: certificate is valid for 172.20.0.1, 10.155.135.90, 172.16.105.174, not 127.0.0.1

Despite that certificate mismatch, _I_, the administrator of the cluster and the tunnel to it, consider this setup secure.
 
The whole point of `insecure-skip-tls-verify` is to say "yes, I know this seems insecure, but I know what I'm doing, let me connect anyway".

I was using Kubectl, k9s, Monokle. They all respect this setting.

Now I wanted to try out Seabird but I'm stuck without any possibility of bypassing this error.

IMHO you should consider also respecting this setting.

**Reply by rasschaert** [reply]
Update: I've managed to proceed with the certificate data in the kubeconfig by adding the actual cluster hostname and pointing it to 127.0.0.1 in my /etc/hosts.

I'm not super happy with this solution but it's a way ahead at least.
